core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1770,9 +1777,10 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
     override def iterator: Iterator[A] =
       left.iterator ++ right.iterator
 
-    override def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit = {
-      left.toArray(n, dest)
-      right.toArray(n + left.length, dest)
+    override protected[zio] def toArray[A1 >: A](srcPos: Int, dest: Array[A1], destPos: Int, length: Int): Unit = {
+      val n = math.max(math.min(math.min(length, left.length - srcPos), dest.length - destPos), 0)
+      left.toArray(srcPos, dest, destPos, n)
```

Comment:
We could skip this call completely if `srcPos > left.length` (the length is needed for the next line anyway)